salt/modules/cp.py

```diff
@@ -8,6 +8,9 @@
 import os
 import logging
 import fnmatch
+import sys
+reload(sys)
+sys.setdefaultencoding('utf-8')
```

Comment:
You'll have to create a _to_unicode function. Then, detect Python 2 using `six.PY2` and convert the specific variables to Unicode.

You can take a look at a PR I have for Nitrogen:
https://github.com/saltstack/salt/pull/40830/files